comp/core/workloadmeta/collectors/internal/kubeapiserver/deployments.go

```diff
@@ -100,14 +107,33 @@ func (p deploymentParser) Parse(obj interface{}) workloadmeta.Entity {
 		}
 	}
 
-	return &workloadmeta.KubernetesDeployment{
+	entities := make([]workloadmeta.Entity, 0, 2)
+
+	deploymentEntity := &workloadmeta.KubernetesDeployment{
 		EntityID: workloadmeta.EntityID{
 			Kind: workloadmeta.KindKubernetesDeployment,
 			ID:   deployment.Namespace + "/" + deployment.Name, // we use the namespace/name as id to make it easier for the admission controller to retrieve the corresponding deployment
 		},
+		EntityMeta: workloadmeta.EntityMeta{
+			Name:        deployment.Name,
+			Namespace:   deployment.Namespace,
+			Labels:      deployment.Labels,
+			Annotations: filterMapStringKey(deployment.Annotations, p.annotationsFilter),
+		},
 		Env:                 deployment.Labels[ddkube.EnvTagLabelKey],
 		Service:             deployment.Labels[ddkube.ServiceTagLabelKey],
 		Version:             deployment.Labels[ddkube.VersionTagLabelKey],
 		InjectableLanguages: containerLanguages,
 	}
+
+	entities = append(entities, deploymentEntity, &workloadmeta.KubernetesMetadata{
+		EntityID: workloadmeta.EntityID{
+			Kind: workloadmeta.KindKubernetesMetadata,
+			ID:   string(util.GenerateKubeMetadataEntityID("apps", "deployments", deployment.Namespace, deployment.Name)),
+		},
+		EntityMeta: deploymentEntity.EntityMeta,
+		GVR:        deployment.GroupVersionKind().GroupVersion().WithResource("deployments"),
+	})
+
+	return entities
```

Comment:
Question ❓ : Since the number of entity if always the same, why not directly do something like:
```
  entities := []workloadmeta.Entity{
    &workloadmeta.KubernetesDeployment{
    EntityID: workloadmeta.EntityID{
        Kind: workloadmeta.KindKubernetesDeployment,
        ID:   deployment.Namespace + "/" + deployment.Name, // we use the namespace/name as id to make it easier for the admission controller to retrieve the corresponding deployment
    },
    EntityMeta: workloadmeta.EntityMeta{
        Name:        deployment.Name,
        Namespace:   deployment.Namespace,
        Labels:      deployment.Labels,
        Annotations: filterMapStringKey(deployment.Annotations, p.annotationsFilter),
    },
    Env:                 deployment.Labels[ddkube.EnvTagLabelKey],
    Service:             deployment.Labels[ddkube.ServiceTagLabelKey],
    Version:             deployment.Labels[ddkube.VersionTagLabelKey],
    InjectableLanguages: containerLanguages,
  },

&workloadmeta.KubernetesMetadata{
    EntityID: workloadmeta.EntityID{
        Kind: workloadmeta.KindKubernetesMetadata,
        ID:   string(util.GenerateKubeMetadataEntityID("apps", "deployments", deployment.Namespace, deployment.Name)),
    },
    EntityMeta: deploymentEntity.EntityMeta,
    GVR:        deployment.GroupVersionKind().GroupVersion().WithResource("deployments"),
},
}
```
something like that ?

Reply:
No specific reason, and I don't think it adds any performance overhead (or am I missing something?)

I already do what you suggested in the pod collector. 

I can update this anyways, thank you for the suggestion 👍

Reply:
No performance things, here the idea is just to keep the code unified (i.e adding the 2 elements at the same time or creating them before and then append them to the slice)
so in the future changes will be easier for other people going through the code. It is mainly for clarity.